src/server/graphql/typesDefs/Filesystem.js

```diff
@@ -98,16 +105,25 @@ export default gql`
         dateCreated: BigInt
         dateModified: BigInt
         type: EntityType
+
+        """
+        The number of times the file has been shared.
+        This is an expensive field, especially when nested inside the contents.
+        """
         shareCount: Int
 
         # Specific to folders
+
+        """
+        The number of files contained in the folder.
+        This is an expensive field, especially when nested inside the contents.
+        """
         fileCount: Int
-    }
 
-    # Allows us to fetch everything at once without having to cache results or
-    # hit stat a bunch of times.
-    type FolderListing {
-        stat: Folder
-        listing: [FilesystemObject]
+        """
+        This an expensive field to include inside content listings.
+        """
+        folderCount: Int
```

Comment:
I almost forgot: the file and folder counts are slow because they come from the stat endpoint, but the listing endpoint does return a `total` field (the combined count for the listing), and that will be required for paging listings in the UI.